### DISCUSSION TITLE
enhance docstring with code examples

### DISCUSSION THREAD

**Discussion by wyli** [Project-MONAI/MONAI#2650] [Ideas]
most pytorch modules come with nice examples in the docstring. for example, the docstring of `torch.nn.modules.Conv1d` [has this block](https://github.com/pytorch/pytorch/blob/f1efe5102876958976b648691ffde422d85b6a0e/torch/nn/modules/conv.py#L178-L188):
```
    Examples::
        >>> m = nn.Conv1d(16, 33, 3, stride=2)
        >>> input = torch.randn(20, 16, 50)
        >>> output = m(input)
    .. _cross-correlation:
        https://en.wikipedia.org/wiki/Cross-correlation
    .. _link:
        https://github.com/vdumoulin/conv_arithmetic/blob/master/README.md
```
these are also available in the autogenerated [documentation site](https://pytorch.org/docs/stable/nn.html#torch.nn.Conv1d): 
![Screenshot 2020-05-30 at 17 43 04](https://user-images.githubusercontent.com/831580/83334252-1723aa80-a29d-11ea-98fa-8682e00f56b7.png)

MONAI's modules should also include such intuitive examples.

(currently It's been done for only a few classes, e.g.:
https://github.com/Project-MONAI/MONAI/blob/master/monai/transforms/croppad/array.py#L164-L176)

**Describe the solution you'd like**
I understand it'll be difficult to enforce this feature for all modules. but once it's there it'll greatly improve usability. this feature request will probably require multiple PRs.

as a starting point, we could extract some examples from the existing unit tests.

**Comment by htudu** [comment]
Hi @wyli,
I would like to work on this. is it open for anyone to contribute?

**Comment by wyli** [comment]
> Hi @wyli,
> I would like to work on this. is it open for anyone to contribute?

yes, it is, please feel free to create PRs (there's a [`contributing.md`](https://github.com/Project-MONAI/MONAI/blob/master/CONTRIBUTING.md) in case you haven't read it)

**Comment by hjmjohnson** [comment]
@wyli  I was planning to have my team work on some of this as we work through using MONAI on our projects.  It is a next item after type hinting is in better shape.

**Comment by hjmjohnson** [comment]
@benjamin-gorman @crtrentz Please keep an eye on this issue, you will be asked to contribute to this effort later this summer.

**Comment by wyli** [comment]
this is a general proposal and has been implemented for many components, I'm converting it to the discussions
